targets/TARGET_Cypress/TARGET_PSOC6/cy_analogout_api.c

```diff
@@ -51,7 +51,7 @@ void analogout_write_u16(dac_t *obj, uint16_t value)
 
 float analogout_read(dac_t *obj)
 {
-    return 100.0f / UINT16_MAX * analogout_read_u16(obj);
+    return analogout_read_u16(obj) / UINT16_MAX;
```

Comment:
This almost always returns 0, as it's integer division. Should be

    return analogout_read_u16(obj) * (1.0f / UINT64_MAX);

(Multiplication by reciprocal will be more efficient than actual FP division).

Same maths should be used in `analogin_read`.